src/app/clusters/testing/CommandTesting.h

```diff
@@ -0,0 +1,269 @@
+    InvokeOperation(const ConcreteCommandPath & path) : mCommandPath(path), mHandler(std::make_unique<MockCommandHandler>())
+    {
+        mRequest.path = mCommandPath;
+    }
+
+    // New constructor: allows injection of a custom MockCommandHandler
+    InvokeOperation(const ConcreteCommandPath & path, std::unique_ptr<MockCommandHandler> handler) :
+        mCommandPath(path), mHandler(std::move(handler))
+    {
+        mRequest.path = mCommandPath;
+    }
+
+    // Optionally, add an overload for (EndpointId, ClusterId, CommandId, handler)
+    InvokeOperation(EndpointId endpoint, ClusterId cluster, CommandId command, std::unique_ptr<MockCommandHandler> handler) :
+        mCommandPath(endpoint, cluster, command), mHandler(std::move(handler))
+    {
+        mRequest.path = mCommandPath;
+    }
+    // Invoke a command with parameters using variadic template
+    // Use for commands that take no parameters (empty TLV structure) or multiple parameters
+    template <typename ClusterType, typename... Args>
+    std::optional<DataModel::ActionReturnStatus> Invoke(ClusterType & cluster, const Args &... args)
+    {
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        TLV::TLVWriter writer;
+        writer.Init(mTlvBuffer);
+
+        // Setup reader (declare before any SuccessOrExit calls)
+        TLV::TLVReader reader;
+
+        if constexpr (sizeof...(args) == 0)
+        {
+            // No parameters - create empty TLV structure
+            TLV::TLVType outerContainer;
+            SuccessOrExit(err = writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outerContainer));
+            SuccessOrExit(err = writer.EndContainer(outerContainer));
+        }
+        else if constexpr (sizeof...(args) == 1)
+        {
+            // Single parameter - use the existing Encode method if available
+            using FirstArg = std::tuple_element_t<0, std::tuple<Args...>>;
+            if constexpr (std::is_same_v<decltype(std::declval<FirstArg>().Encode(std::declval<TLV::TLVWriter &>(),
+                                                                                  std::declval<TLV::Tag>())),
+                                         CHIP_ERROR>)
+            {
+                // It's an encodable object
+                SuccessOrExit(err = std::get<0>(std::tie(args...)).Encode(writer, TLV::AnonymousTag()));
+            }
+            else
+            {
+                // It's a primitive type, encode as structure with single field
+                TLV::TLVType outerContainer;
+                SuccessOrExit(err = writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outerContainer));
+                SuccessOrExit(err = DataModel::Encode(writer, TLV::ContextTag(0), std::get<0>(std::tie(args...))));
+                SuccessOrExit(err = writer.EndContainer(outerContainer));
+            }
+        }
+        else
+        {
+            // Multiple parameters - encode as structure with multiple fields
```

Comment:
Don't we have request structures?

like looking at https://github.com/project-chip/connectedhomeip/blob/master/zzz_generated/app-common/clusters/DoorLock/Commands.h#L194 that contains both encode and decode and has information about command and cluster id (so that we do not need to provide that)